ra/ra.go

```diff
@@ -2168,40 +2168,54 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 			minExpiry = authzExpiry
 		}
 	}
-
-	// If new authorizations are needed, call AddPendingAuthorizations. Also check
-	// whether the newly created pending authz's have an expiry lower than minExpiry
+	// If the newly created pending authz's have an expiry closer than the
+	// minExpiry the minExpiry is the pending authz expiry.
 	if len(newAuthzs) > 0 {
-		req := sapb.AddPendingAuthorizationsRequest{Authz: newAuthzs}
-		authzIDs, err := ra.SA.NewAuthorizations2(ctx, &req)
+		newPendingAuthzExpires := ra.clk.Now().Add(ra.pendingAuthorizationLifetime)
+		if newPendingAuthzExpires.Before(minExpiry) {
+			minExpiry = newPendingAuthzExpires
+		}
+	}
+	// Set the order's expiry to the minimum expiry. The db doesn't store
+	// sub-second values, so truncate here.
+	order.Expires = minExpiry.Truncate(time.Second).UnixNano()
+
+	var storedOrder *corepb.Order
+	if features.Enabled(features.StreamlineOrderAndAuthzs) {
+		newOrderAndAuthzsReq := &sapb.NewOrderAndAuthzsRequest{
+			Order:     order,
+			NewAuthzs: newAuthzs,
+		}
+		storedOrder, err = ra.SA.NewOrderAndAuthzs(ctx, newOrderAndAuthzsReq)
```

Comment:
Should probably contain the same kinds of order validation checks present in #5615